doc/specs/#1337 - Per-Profile Tab Colors/#1337 - Per-Profile Tab Colors.md

```diff
@@ -0,0 +1,149 @@
+continue working when we add support for themes in the future.
+
+## Solution Design
+
+To enable per-profile tab colors, we'll add a single setting: `tabColor`. For
+now<sup>[[1](#user-content-footnote-1)]</sup>, this setting will accept any
+`#rrggbb` color string.
+
+Since each profile creates a `Pane` with a `TermControl`, we'll need to store
+this color not in the `Tab`, but somewhere below `Pane`, so that when you switch
+between Panes in a tab with different `tabColor`s, the color will update
+automatically. When a new `TermControl` is created, we'll store this color in the
+`TermControl`'s `Terminal` core. This is to support the future possibility of
+setting the tab color via VT sequences.
+
+A Tab's color will be the result of layering a variety of sources, from the bottom up:
+
+Color | | Set by
+-- | -- | --
+Runtime Color | _optional_ |Color Picker / `setTabColor` action
+Control Tab Color | _optional_ | Profile's `tabColor`, or a color set by VT
+Theme Tab Background | _optional_ | `tab.backgroundColor` in the theme
+Tab Default Color | **default** | TabView in XAML
+
+Some examples:
+* **Scenario 1**: The user has set `"tabColor": "#ff0000"` in their profile.
+  When they create tabs with that profile, instead of appearing in the default
+  color for the TabView, they'll be `#ff0000`.
+* **Scenario 2**: The user has set `"tabColor": "#ff0000"` in their profile.
+  When they try to set the color for that tab (with the color picker) to
+  `#0000ff`, the tab's color is updated to reflect this new blue color. When
+  they clear the runtime color (with the color picker), the tab will return to
+  `#ff0000`.
+* **Scenario 3**: The user has two profiles with colors set, one to `"tabColor":
+  "#ff0000"`, and the other with `"tabColor": "#00ff00"`. If they open these
+  profiles in two panes side-by side, the tab's color will update to reflect the
+  color from the currently-focused control.
+* **Scenario 4**: The user has two profiles with colors set, one to `"tabColor":
+  "#ff0000"`, and the other with `"tabColor": "#00ff00"`. If they open these
+  profiles in two panes side-by side, and try to set the color for that tab
+  (with the color picker) to `#0000ff`, the tab's color is updated to reflect
+  this new blue color. Regardless of which pane is focused, the tab will be
+  blue.
+* **Scenario 5**: The user has set `"tabColor": "#ff0000"` in their profile
+  ("Profile A"), and `"tab.backgroundColor": "#00ff00"`in their theme. When they
+  create tabs with "Profile A", the tabs will appear red. Other tabs (for
+  profiles without `tabColor` set) will appear green, using the color from the
+  theme.
+
+
+## UI/UX Design
+
+In general, this is going to look exactly like the colored tabs look now.
+
+**TODO** Should using the tab color picker to clear a tab's color clear the
+currently set color? E.g if there's a Control Color, but no Runtime Color,
+should clear color remove the Control color to revert to the theme/default
+color?

```

Comment:
2 things (this is all opinion though so I could very much be wrong):
1. "clearing the color" (represented by setting the color to 'null') just means to fallback according to your chart above.
2. The tab color picker and all ways of invoking it (i.e.: context menu, keybinding, etc...) needs to present the fallback as an option. I imagine it being something like this...
   - `profileColor`: fall back to the profile's color
   - `themeColor`: fall back to the theme's color
   - I don't think TabView color should even be an option tbh.  I feel like the absence of a theme is in itself a theme.

Reply:
Yea I was more thinking does "clear" mean "clear whatever the current set color is, regardless of where it came from", or does it mean "clear the runtime color". So clear with a runtime color would take it to the composition of Control/Theme/Default, but clearing it from there (without a runtime color) would clear whatever the next set layer is. idk, typing it out now, that sounds horribly convoluted

I mostly included the TabView default to make it clear that that is the default value. I suppose if the theme value is `#00ff00`, then there doesn't _really_ need to be a "tab default" - that _is_ the tab default.

Reply:
You know what, I'm just yanking this section for now. It's too confusing, and I think this is something to discuss down the line (if we want)